compiler/rustc_session/src/config.rs

```diff
@@ -1999,6 +2029,8 @@ pub fn build_session_options(matches: &getopts::Matches) -> Options {
 
     let incremental = cg.incremental.as_ref().map(PathBuf::from);
 
+    let assert_incr_state = parse_assert_incr_state(matches);
```

Comment:
Instead of retrieving the string value from `matches`, you can get it from `debugging_opts` via the field you declared.